scripts/verify-spelling.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TOOL_VERSION="v0.3.4"
+
+LANGUAGE="${1:-en}"
+# cd to the root path
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+cd "${ROOT}"
+
+# create a temporary directory
+TMP_DIR=$(mktemp -d)
+
+# cleanup
+exitHandler() (
+  echo "Cleaning up..."
+  rm -rf "${TMP_DIR}"
+)
+trap exitHandler EXIT
+
+# perform go get in a temp dir as we are not tracking this version in a go module
+# if we do the go get in the repo, it will create / update a go.mod and go.sum
+cd "${TMP_DIR}"
+GO111MODULE=on GOBIN="${TMP_DIR}" go get "github.com/client9/misspell/cmd/misspell@${TOOL_VERSION}"
+export PATH="${TMP_DIR}:${PATH}"
+cd "${ROOT}"
+
+# check spelling
+RES=0
+echo "Checking spelling..."
+ERROR_LOG="${TMP_DIR}/errors.log"
+git ls-files | grep content/${LANGUAGE} | xargs misspell > "${ERROR_LOG}"
```

Comment:
Hi @rikatz .
I did not get a chance to run the script, but the code looks okay.
It would be good to target any Markdown files (.md files), especially files in `content/en/docs`.
I am not familiar with this spellchecker. Does the tool report a lot of false positives?
Thanks!

Reply:
Hi @kbhawkey 

The main problem of targeting all the MDs is that unfortunately the spell check only works with english right now :/ I couldn't find a way for other languages other than using aspell and being really verbose with false positives (like we would have to add a lot of tech words). But I can do this as a follow up for sure! 

Python does have this multi language support (I've found some in the past) and can take a look again, and maybe have an opt-in. [here](https://github.com/rojopolis/spellcheck-github-actions) is one example made with pyspellcheck as a GH action. 

The original proposal was to have this as an optional script to be run by folks making PRs. Also I didn't targeted blog contents and auto generated files to not change things that can be rolled back later by some automatic process.

About the output and the tool, actually what it does is it have the most common mispells from English language (and they are a lot) and generate which files have them and the correction, as an example let me put an output here from before my PR #26229 

```
./scripts/verify-spelling.sh 
go: downloading github.com/client9/misspell v0.3.4
go: found github.com/client9/misspell/cmd/misspell in github.com/client9/misspell v0.3.4
Checking spelling...
error: content/en/blog/_posts/2016-05-00-Coreosfest2016-Kubernetes-Community.md:15:189: "Revelant" is a misspelling of "Relevant"
error: content/en/blog/_posts/2016-10-00-Dynamic-Provisioning-And-Storage-In-Kubernetes.md:146:203: "existance" is a misspelling of "existence"
error: content/en/blog/_posts/2017-02-00-Postgresql-Clusters-Kubernetes-Statefulsets.md:44:5: "exportfs" is a misspelling of "exports"
error: content/en/blog/_posts/2018-06-28-Airflow-Kubernetes-Operator.md:195:444: "contributers" is a misspelling of "contributors"
error: content/en/blog/_posts/2018-10-02-network-bootable-farm-with-ltsp.md:463:20: "netowrk" is a misspelling of "network"
error: content/en/blog/_posts/2019-03-19-kubeedge-k8s-based-edge-intro.md:56:119: "mosquitto" is a misspelling of "mosquito"
error: content/en/blog/_posts/2020-05-06-third-party-dual-sourced-content.md:30:86: "continous" is a misspelling of "continuous"
error: content/en/blog/_posts/2020-05-21-wsl2-dockerdesktop-k8s.md:523:19: "embeded" is a misspelling of "embedded"
error: content/en/blog/_posts/2020-07-27-kubernetes-1-17-release-interview.md:201:145: "contributer" is a misspelling of "contributor"
error: content/en/docs/setup/best-practices/multiple-zones.md:32:29: "interchangable" is a misspelling of "interchangeable"
error: content/en/docs/reference/using-api/api-concepts.md:517:33: "availabiliy" is a misspelling of "availability"
error: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:276:67: "availabe" is a misspelling of "available"
error: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:637:21: "requried" is a misspelling of "required"
error: content/en/docs/tasks/administer-cluster/dns-custom-nameservers.md:258:31: "offical" is a misspelling of "official"
error: content/en/docs/tasks/administer-cluster/topology-manager.md:195:308: "explicity" is a misspelling of "explicitly"
error: content/en/docs/tasks/configure-pod-container/configure-gmsa.md:257:227: "recieve" is a misspelling of "receive"
error: content/en/docs/setup/release/notes.md:131:37: "seperating" is a misspelling of "separating"
error: content/en/docs/setup/release/notes.md:193:161: "infromation" is a misspelling of "information"
error: content/en/docs/setup/release/notes.md:193:211: "exprimental" is a misspelling of "experimental"
error: content/en/docs/setup/release/notes.md:395:114: "extention" is a misspelling of "extension"
error: content/en/docs/setup/release/notes.md:1575:114: "extention" is a misspelling of "extension"
error: content/en/docs/setup/release/notes.md:1704:114: "extention" is a misspelling of "extension"
Found spelling errors!
Cleaning up...
```

Reply:
Just to followup with the Python idea:

* Need to install aspell on the container that will run that (and the dictionaries)
* Install the pyspelling pip
* Create some wordlist (like kubectl, kubernetes, kubelet, localhost, sig, yaml, etc etc)
* Create a config file (simple but can be evolved, as):
```
matrix:
- name: Kubernetes EN Doc 
  aspell:
    lang: en
    d: en_US
  sources:
  - content/en/**/*.md
pipeline:
  - pyspelling.filters.markdown:
```
* Run pyspelling -c spellcheck.yaml

And it outputs like this:
```
Misspelled words:
<text> content/en/docs/contribute/localization.md
--------------------------------------------------------------------------------
Approvers
BPD
CNCF
Deutsch
ICH
Kubernetes
Localizations
Minikube
```

But as you can see it's going to be a huge effort to ignore tech words :)

Reply:
Thanks for investigating @rikatz .